File: abstractions/php/src/HttpMethod.php
<?php
namespace Microsoft\Kiota\Abstractions;

final class HttpMethod extends Enum {
    public const GET = "GET";
    public const POST = "POST";
    public const PATCH = "PATCH";
    public const DELETE = "DELETE";
    public const OPTIONS = "OPTIONS";
    public const CONNECT = "CONNECT";
    public const TRACE = "TRACE";
    public const HEAD = "HEAD";
    public const PUT = "PUT";
}

File: abstractions/dotnet/src/HttpMethod.cs
namespace Microsoft.Kiota.Abstractions {
    /// <summary>
    ///     Represents the HTTP method used by a request.
    /// </summary>
    public enum HttpMethod {
        /// <summary>
        ///     The HTTP GET method.
        /// </summary>
        GET,
        /// <summary>
        ///     The HTTP POST method.
        /// </summary>
        POST,
        /// <summary>
        ///     The HTTP PATCH method.
        /// </summary>
        PATCH,
        /// <summary>
        ///     The HTTP DELETE method.
        /// </summary>
        DELETE,
        /// <summary>
        ///     The HTTP OPTIONS method.
        /// </summary>
        OPTIONS,
        /// <summary>
        ///     The HTTP PUT method.
        /// </summary>
        PUT,
        /// <summary>
        ///     The HTTP HEAD method.
        /// </summary>
        HEAD,
        /// <summary>
        ///     The HTTP CONNECT method.
        /// </summary>
        CONNECT,
        /// <summary>
        ///     The HTTP TRACE method.
        /// </summary>
        TRACE
    }
}
